### PR TITLE
Apple TV Support:

### PR DESCRIPTION
**Pull request by samrobbins**
- Stores appletvos screen shots in a subdirectory.
- Detects appletvos screen files correctly.
- Update help text to include appleTV instructions.

### CODE REVIEW

**Comment by samrobbins** [comment]
This requires a commit in `spaceship` to function correctly. https://github.com/fastlane/spaceship/commit/f17033e49f8a66f88fbeadf3c4e4b2cda1c636c3

**Comment by samrobbins** [comment]
It is expected that CI will break as `spaceship` was not deployed yet with my changes.

**Comment by KrauseFx** [comment]
@samrobbins You can deploy the new version of `spaceship` and update the `spaceship` dependency of `deliver` :rocket:

**Comment by samrobbins** [comment]
@KrauseFx this is green now :balloon:

File: lib/deliver/app_screenshot.rb [hunk 1]
```diff
@@ -129,14 +129,35 @@ def self.calculate_screen_size(path)
...
+
+    def self.calculate_screen_size(path)
+      size = FastImage.size(path)
+
+      raise "Could not find or parse file at path '#{path}'" if size.nil? or size.count == 0
+
+      # Walk up two directories and test if we need to handle a platform that doesn't support landscape
+      path_component = Pathname.new(path).each_filename.to_a[-3]
+      if path_component.eql? "appleTV"
+        skip_landscape = true
```
Comment (KrauseFx) [inline lib/deliver/app_screenshot.rb]: :+1:

File: lib/deliver/loader.rb [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 module Deliver
   module Loader
-    ALL_LANGUAGES = FastlaneCore::Languages::ALL_LANGUAGES.map(&:downcase).freeze
+    # The directory 'appleTV' is a special folder that will cause our screenshot gathering code to iterate
+    # through it as well searching for language folders.
+    APPLE_TV_DIR_NAME = "appleTV"
+    ALL_LANGUAGES = (FastlaneCore::Languages::ALL_LANGUAGES + [APPLE_TV_DIR_NAME]).map(&:downcase).freeze
```
Comment (KrauseFx) [inline lib/deliver/loader.rb]: :+1:

File: lib/deliver/upload_screenshots.rb [hunk 3]
```diff
@@ -54,21 +54,33 @@ def upload(options, screenshots)
...
       screenshots = []
       extensions = '{png,jpg,jpeg}'
-      Loader.language_folders(options[:screenshots_path]).each do |lng_folder|
+
+      Loader.language_folders(path).each do |lng_folder|
+        language = File.basename(lng_folder)
+
+        # Check to see if we need to traverse multiple platforms or just a single platform
+        if language == Loader::APPLE_TV_DIR_NAME
+          screenshots.concat collect_screenshots_for_languages(File.join(path, language))
```
Comment (KrauseFx) [inline lib/deliver/upload_screenshots.rb]: I'd add `(` around the `concat` call to make it more clear :+1:

**Comment by KrauseFx** [comment]
This looks great already @samrobbins, hard problem - great solution. Could you also update the documentation to explain where the tv screenshots should be stored? 

Does `deliver init` automatically create the folder + the tv screenshots?

**Comment by samrobbins** [comment]
@KrauseFx deliver init does the correct thing with existing screenshots. Let me test with no screenshots.

**Comment by samrobbins** [comment]
If no screenshots are available to download no appleTV folder is created. Should we try to auto detect this and create it? Saying that a project is tvOS only is hard since it could be a multi target project. Is updating the documentation good enough here?

**Comment by samrobbins** [comment]
@KrauseFx `(` add to concat call as well.

**Comment by KrauseFx** [comment]
LGTM, testing it on some real-world project before it's ready to be merged :+1:

**Comment by KrauseFx** [comment]
Looks great, :ship:

**Pull request merged**